src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,159 @@
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getLogin(), 6),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user password length less then 6 characters")
+    void register_UserPasswordLengthLessThenSixCharacters_notOk() {
+        newUser.setLogin("valid login");
+        newUser.setPassword("test");
+        expectedMessage = "User password %s length less then %d characters";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getPassword(), 6),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user age less then 18")
+    void register_UserAgeLessThenEighteen_notOk() {
+        newUser.setLogin("valid login");
+        newUser.setPassword("valid password");
+        newUser.setAge(17);
+        expectedMessage = "User age %d less then %d years";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getAge(), 18),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user with such login already exists")
+    void register_UserAlreadyExists_notOk() {
+        User oldUser = new User(1L, "valid login", "valid password", 18);
+        storageDao.add(oldUser);
+        newUser = new User(2L, "valid login", "other valid password", 18);
+        expectedMessage = "User with login %s already exists";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                String.format(expectedMessage, newUser.getLogin()),
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    void register_registerNewUser_ok() {
+        User expectedUser = new User(1L, "test login", "test password", 18);
+        User actualUser = new User(1L, "test login", "test password", 18);
+        int expectedSize = Storage.people.size() + 1;
+        assertAll(
+                () -> assertEquals(expectedUser, registrationService.register(actualUser)),
+                () -> assertEquals(expectedSize, Storage.people.size())
+        );
```

Comment:
Redundant complication
Also let's add the check for the actual user in the storage
```suggestion
            assertEquals(expectedUser, registrationService.register(actualUser)),
            assertEquals(expectedSize, Storage.people.size())
            // storage contains our user assertion
```